### ISSUE TITLE
Page bugs out when nearing the end of a chapter 

### ISSUE THREAD

**Issue by FeeninForMore** [KotatsuApp/Kotatsu#951]
### Brief summary

When I'm near the end of a chapter the page bugs out and a cutout page appears at the bottom 

### Steps to reproduce

I don't really know how to reproduce it, it's been happening to me everytime 

### Kotatsu version

7.2.1

### Android version

14

### Device

Pixel 8a

### Acknowledgements

- [X] This is not a duplicate of an existing issue. Please look through the list of [open issues](https://github.com/KotatsuApp/Kotatsu/issues) before creating a new one.
- [X] This is not an issue with a specific manga source. Otherwise, you have to open an issue in the [parsers repository](https://github.com/KotatsuApp/kotatsu-parsers/issues/new/choose).

**Comment by FeeninForMore** [comment]
![a1bdabe9-7421-425c-bcdf-ab52ac58225c](https://github.com/KotatsuApp/Kotatsu/assets/123899765/fc762a15-b292-47c9-9d69-00f4ca4bd466)

**Comment by FeeninForMore** [comment]
It's actually kind of annoying it stays on the screen until you swipe up

**Comment by sleepingzzs** [comment]
it has happened to me too. ive noticed this usually occurs when you zoom in and out

**Comment by FeeninForMore** [comment]
I reported this on the telegram server and they tried saying i was scrolling too fast 💀

**Comment by Raja-Debnath** [comment]
I thought this issue was due to the manga provider side , I faced it on mangafire

**Comment by FeeninForMore** [comment]
Na it happens with all sources

**Comment by Raja-Debnath** [comment]
@FeeninForMore 
Any idea how this can be fixed? 

Also I'm not app dev , what technology is the app made of flutter?

**Comment by FeeninForMore** [comment]
I have no clue how to fix it, better just to wait for the Dev to comment

**Comment by galpt** [comment]
There are some things that I could think of that could cause this to happen:
1. Your phone has small memory (RAM), my phone has only 8 GB but that's still small for Android.
2. It's because some images are slow to load. Could be avoided by using the Img Optimization Proxy feature. I keep on recommending people to use it since weeks ago.
3. It's probably an issue with the Kotatsu reader, but I doubt that. Point (1) and (2) are what likely to cause this IMHO.

Point (1) can be the cause because any apps need memory to display images.
Reading manga can consume your phone memory quickly without you even realizing it.
It's easier to explain it if you're a developer/software engineer or you happen to care about memory usage on your devices (either laptop or phone).
There's no quick solution other than getting a new more powerful phone, or just use Kotatsu on an emulator (i.e. BlueStacks).

Point (2) is very likely to happen. Most manga websites host their images on free image hosting services.
Those services can be slow because of overloaded servers, or it's just you getting rate limited by the service.
You should use the img proxy feature whenever possible, since one of the reasons is to prevent readers from getting rate limited.

Point (3) is what I think the chance is slim.
I believe point (1) or (2) is the most logical reason since both can be explained technically, but I also suspect point (3) can be the cause.

**Comment by FeeninForMore** [comment]
I'm not gonna say the pixel 8a is the fastest android phone, but it's not slow I'm 99% it's not my phone's problem

**Comment by FeeninForMore** [comment]
But I will use the IMG proxy feature ig

**Comment by FeeninForMore** [comment]
Also this problem only appears on kotatsu, it works fine  on mihon

**Comment by galpt** [comment]
Note that long colorful images like what you see in manga, they can be as big as 2 - 7 MB I've seen per-image.
I'm not saying people's phones can't handle them, but loading these images can eat up phone memory real quick.

One thing that people can try immediately (almost like a quick solution if it works) is by using the img proxy and see if this issue happens less than when you're not using the proxy.

**Comment by galpt** [comment]
> Also this problem only appears on kotatsu, it works fine on mihon

Yeah that's why I mentioned that point (3) is still possible. It might be Kotatsu reader that caused it.
But I'm curious if this problem still happens that often after you're using the img proxy feature.
